API-Samples/immutable_sampler/immutable_sampler.cpp

```diff
@@ -273,7 +273,11 @@ int sample_main(int argc, char *argv[]) {
     vkDestroySampler(info.device, immutableSampler, NULL);
     vkDestroyImageView(info.device, info.textures[0].view, NULL);
     vkDestroyImage(info.device, info.textures[0].image, NULL);
-    vkFreeMemory(info.device, info.textures[0].mem, NULL);
+    vkFreeMemory(info.device, info.textures[0].image_memory, NULL);
+    if (info.textures[0].needs_staging) {
+        vkDestroyBuffer(info.device, info.textures[0].buffer, NULL);
```

Comment:
should we set info.textures[0].buffer to VK_NULL_HANDLE to avoid a double delete?